Products

koikos opened the pull request:
CRUD operations on Products.
When running locally, navigate to localhost:3000/products

koikos committed:
Generate MVC components for products

koikos committed:
Add READ opearation for products

koikos committed:
Fix validation rule to allow 0 product units in stock

koikos committed:
Add CREATE, UPDATE, DELETE operations for products

klaudia-janiec commented on app/controllers/products_controller.rb:
```diff
@@ -0,0 +1,71 @@
+  def create
+    @product = Product.new
+    @product.assign_attributes(permitted_attributes)
+    @product.assign_attributes(units_on_order: 0)
```
Skoro to nadpisujemy to dlaczego w ogóle dopuszczamy ten parametr w permitted attributes? Nie widzę aby go można było wysłać także przez formularz. Czy to znaczy ile ogólnie zamówiono jednostek danego produktu? Jeśli tak to powinniśmy usunąć ten atrybut z `permitted_attributes`. Przydałoby się też zwiększać tą wartość z każdym złożonym zamówieniem na dany produkt. Ale to można zrobić już potem. :wink:

koikos replied:
Usunąłem `units_on_order` - przeoczenie.
Istotnie, przy składaniu zamówienia należałoby zmieniać wartość tego parametru, ale tu by trzeba się zastanowić nad logiką - wydaje mi się, że powinien on odzwierciedlać stan, kiedy produkt nie jest jeszcze wysłany (jest w magazynie), ale już jest przypisany do zamówienia. Myślę, że na tym etapie nie będziemy tego zmieniać.

klaudia-janiec reviewed (approved):
Jest OK, zwróć uwagę tylko na `units_on_order`.  :wink:

koikos committed:
Romove unnecessary 'units_on_order' from permitted_attributes

'units_on_order' is an atrribute which holds amount of units of given
product that was already ordered. When creating new product its value
is obviously 0, and when modyfing product data manually we do not want
to change this value not to make the data inconsistent. This is why
the crete/update form does not have corresponding field. This makes
'units_on_order' attribute in 'permitted_attributes' unneeded.
